In [1]:
import warnings
warnings.filterwarnings("ignore")
import intake

In [2]:
col = intake.open_esm_metadatastore(collection_input_definition="CESM1-LE")

Getting file listing: CTRL:GLADE:posix:/glade/collections/cdg/data/cesmLE/CESM-CAM5-BGC-LE


Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.THBOT.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.ERRH2OSNO.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.PSNSUN.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.PRODUCT_CLOSS.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.PSNSUN_TO_CPOOL.200601-201412.nc_temp_.nc does not conform to expected pattern

Getting file listing: CTRL:HPSS:hsi:/CCSM/csm/CESM-CAM5-BGC-LE
Getting file listing: 20C:GLADE:posix:/glade/collections/cdg/data/cesmLE/CESM-CAM5-BGC-LE


Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.THBOT.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.ERRH2OSNO.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.PSNSUN.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.PRODUCT_CLOSS.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.PSNSUN_TO_CPOOL.200601-201412.nc_temp_.nc does not conform to expected pattern

Getting file listing: 20C:HPSS:hsi:/CCSM/csm/CESM-CAM5-BGC-LE
Getting file listing: RCP85:GLADE:posix:/glade/collections/cdg/data/cesmLE/CESM-CAM5-BGC-LE


Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.THBOT.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.ERRH2OSNO.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.PSNSUN.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.PRODUCT_CLOSS.200601-201412.nc_temp_.nc does not conform to expected pattern
Filename : f.e11.FAMIPC5CN.f09_f09.rcp85.toga.ens10.clm2.h0.PSNSUN_TO_CPOOL.200601-201412.nc_temp_.nc does not conform to expected pattern

Getting file listing: RCP85:HPSS:hsi:/CCSM/csm/CESM-CAM5-BGC-LE
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110702 entries, 0 to 110701
Data columns (total 18 columns):
resource            110702 non-null object
resource_type       110702 non-null object
direct_access       110702 non-null object
experiment          110702 

In [3]:
col.df.head()

resource resource_type  \
0  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
1  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
2  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
3  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
4  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   

   direct_access experiment                         case component  stream  \
0           True       CTRL  b.e11.B1850C5CN.f09_g16.005       atm  cam.h1   
1           True       CTRL  b.e11.B1850C5CN.f09_g16.005       atm  cam.h1   
2           True       CTRL  b.e11.B1850C5CN.f09_g16.005       atm  cam.h1   
3           True       CTRL  b.e11.B1850C5CN.f09_g16.005       atm  cam.h1   
4           True       CTRL  b.e11.B1850C5CN.f09_g16.005       atm  cam.h1   

  variable         date_range  member_id  \
0     T200  18000101-18991231          0   
1     T200  08000101-08991231          0   
2     T200  04020101-04991231          0   
3     T200  10000101-10991231          0   
4     T200  11000101-11991231          0   

                                       file_fullpath  \
0  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
1  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
2  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
3  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
4  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   

                                       file_basename  \
0  b.e11.B1850C5CN.f09_g16.005.cam.h1.T200.180001...   
1  b.e11.B1850C5CN.f09_g16.005.cam.h1.T200.080001...   
2  b.e11.B1850C5CN.f09_g16.005.cam.h1.T200.040201...   
3  b.e11.B1850C5CN.f09_g16.005.cam.h1.T200.100001...   
4  b.e11.B1850C5CN.f09_g16.005.cam.h1.T200.110001...   

                                        file_dirname  ctrl_branch_year  \
0  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
1  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
2  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
3  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
4  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   

   year_offset  sequence_order  has_ocean_bgc grid  
0       1448.0               0           True  NaN  
1       1448.0               0           True  NaN  
2       1448.0               0           True  NaN  
3       1448.0               0           True  NaN  
4       1448.0               0           True  NaN

In [4]:
col.df.resource_type.unique()

array(['posix'], dtype=object)

In [13]:
variables = col.df.variable.unique()

## List of variables to include in AWS

In [6]:
atmos_mon_3d = ["T", "U", "V", "Q", "Z3"]


### Atmosphere, monthly, 3D

In [11]:
cat1 = col.search(variable=atmos_mon_3d, component="atm")
cat1.query_results.variable.unique()

array(['Q', 'T', 'U', 'V', 'Z3'], dtype=object)

In [12]:
cat1.query_results.head()

resource resource_type  \
1077  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
1073  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
1072  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
1068  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   
1078  CTRL:GLADE:posix:/glade/collections/cdg/data/c...         posix   

      direct_access experiment                         case component  stream  \
1077           True       CTRL  b.e11.B1850C5CN.f09_g16.005       atm  cam.h0   
1073           True       CTRL  b.e11.B1850C5CN.f09_g16.005       atm  cam.h0   
1072           True       CTRL  b.e11.B1850C5CN.f09_g16.005       atm  cam.h0   
1068           True       CTRL  b.e11.B1850C5CN.f09_g16.005       atm  cam.h0   
1078           True       CTRL  b.e11.B1850C5CN.f09_g16.005       atm  cam.h0   

     variable     date_range  member_id  \
1077        Q  040001-049912          0   
1073        Q  050001-059912          0   
1072        Q  060001-069912          0   
1068        Q  070001-079912          0   
1078        Q  080001-089912          0   

                                          file_fullpath  \
1077  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
1073  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
1072  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
1068  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   
1078  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...   

                                          file_basename  \
1077  b.e11.B1850C5CN.f09_g16.005.cam.h0.Q.040001-04...   
1073  b.e11.B1850C5CN.f09_g16.005.cam.h0.Q.050001-05...   
1072  b.e11.B1850C5CN.f09_g16.005.cam.h0.Q.060001-06...   
1068  b.e11.B1850C5CN.f09_g16.005.cam.h0.Q.070001-07...   
1078  b.e11.B1850C5CN.f09_g16.005.cam.h0.Q.080001-08...   

                                           file_dirname  ctrl_branch_year  \
1077  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
1073  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
1072  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
1068  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   
1078  /glade/collections/cdg/data/cesmLE/CESM-CAM5-B...               NaN   

      year_offset  sequence_order  has_ocean_bgc grid  
1077       1448.0               0           True  NaN  
1073       1448.0               0           True  NaN  
1072       1448.0               0           True  NaN  
1068       1448.0               0           True  NaN  
1078       1448.0               0           True  NaN

In [20]:
cat1.query_results.date_range.unique()

array(['040001-049912', '050001-059912', '060001-069912', '070001-079912',
       '080001-089912', '090001-099912', '100001-109912', '110001-119912',
       '120001-129912', '130001-139912', '140001-149912', '150001-159912',
       '160001-169912', '170001-179912', '180001-189912', '190001-199912',
       '200001-209912', '210001-220012', '185001-200512', '192001-200512',
       '200601-208012', '208101-210012', '200601-210012'], dtype=object)

In [15]:
cat1.query_results.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 655 entries, 1077 to 49287
Data columns (total 18 columns):
resource            655 non-null object
resource_type       655 non-null object
direct_access       655 non-null bool
experiment          655 non-null object
case                655 non-null object
component           655 non-null object
stream              655 non-null object
variable            655 non-null object
date_range          655 non-null object
member_id           655 non-null int64
file_fullpath       655 non-null object
file_basename       655 non-null object
file_dirname        655 non-null object
ctrl_branch_year    565 non-null float64
year_offset         90 non-null float64
sequence_order      655 non-null int64
has_ocean_bgc       655 non-null bool
grid                0 non-null object
dtypes: bool(2), float64(2), int64(2), object(12)
memory usage: 88.3+ KB


**NOTE:**
    
- There are **655** files. The spreadsheet mentions existance of **1010** files. 

### Atmosphere, Monthly, Surface

In [16]:
atmos_mon_surface = ["FLNS", "FLNSC", "FLUT", "FSNS", "FSNSC", "FSNTOA", "ICEFRAC", 
                     "LHFLX", "PRECC", "PRECL", "PRECSC", "PRECSL", "PSL", "SHFLX", 
                     "TMQ", "TREFHT", "TREFHTMN", "TREFHTMX", "TS"]

In [18]:
cat2 = col.search(variable=atmos_mon_surface, component="atm")
cat2.query_results.variable.unique(), cat2.query_results.variable.nunique()

(array(['FLNS', 'FLNSC', 'FLUT', 'FSNS', 'FSNSC', 'FSNTOA', 'ICEFRAC',
        'LHFLX', 'PRECL', 'PRECSC', 'PRECSL', 'PSL', 'SHFLX', 'TMQ',
        'TREFHT', 'TREFHTMN', 'TREFHTMX', 'TS', 'PRECC'], dtype=object), 19)

In [19]:
cat2.query_results.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4848 entries, 224 to 49409
Data columns (total 18 columns):
resource            4848 non-null object
resource_type       4848 non-null object
direct_access       4848 non-null bool
experiment          4848 non-null object
case                4848 non-null object
component           4848 non-null object
stream              4848 non-null object
variable            4848 non-null object
date_range          4848 non-null object
member_id           4848 non-null int64
file_fullpath       4848 non-null object
file_basename       4848 non-null object
file_dirname        4848 non-null object
ctrl_branch_year    4182 non-null float64
year_offset         666 non-null float64
sequence_order      4848 non-null int64
has_ocean_bgc       4848 non-null bool
grid                0 non-null object
dtypes: bool(2), float64(2), int64(2), object(12)
memory usage: 653.3+ KB


In [21]:
cat2.query_results.date_range.unique()

array(['04020101-04991231', '05000101-05991231', '06000101-06991231',
       '07000101-07991231', '08000101-08991231', '09000101-09991231',
       '10000101-10991231', '11000101-11991231', '12000101-12991231',
       '13000101-13991231', '14000101-14991231', '15000101-15991231',
       '16000101-16991231', '17000101-17991231', '18000101-18991231',
       '19000101-19991231', '20000101-20991231', '21000101-22001231',
       '18500101-20051231', '19200101-20051231', '040001-049912',
       '050001-059912', '060001-069912', '070001-079912', '080001-089912',
       '090001-099912', '100001-109912', '110001-119912', '120001-129912',
       '130001-139912', '140001-149912', '150001-159912', '160001-169912',
       '170001-179912', '180001-189912', '190001-199912', '200001-209912',
       '210001-220012', '185001-200512', '192001-200512',
       '20060101-20801231', '20810101-21001231', '20060101-21001231',
       '200601-208012', '208101-210012', '200601-210012'], dtype=object)